printbuf.c

```diff
@@ -65,6 +66,8 @@ static int printbuf_extend(struct printbuf *p, int min_size)
 
 	if (p->size >= min_size)
 		return 0;
+	if (p->size > INT32_MAX / 2 || min_size > INT32_MAX - 8)
```

Comment:
Let's not limit the size to `INT32_MAX / 2`.  Instead, if we're above that threshold then instead of doing `new_size = p->size * 2` just always set it to `min_size + 8`.

Reply:
Code is a bit more complex but should fit these requirements now.